src/backend/cdb/cdbpath.c

```diff
@@ -225,17 +241,25 @@ cdbpath_create_motion_path(PlannerInfo *root,
 			return (Path *) pathnode;
 		}
 
-		/* replicated-->singleton would give redundant copies of the rows. */
+		/*
+		 * replicated-->singleton would give redundant copies of the rows.
+		 * But there is an exception - DML over replicated table should pass
+		 * the check and add motion.
```

Comment:
The similar question. We know that sometimes locus Replicated may be successfully cast to Bottleneck (by focusing with Gather Motion). What sort of plans we want to filter here if they will be handled properly?